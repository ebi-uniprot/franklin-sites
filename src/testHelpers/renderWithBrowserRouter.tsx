import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

// Inspired by https://testing-library.com/docs/example-react-router/

const renderWithBrowserRouter = (ui: ReactElement, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
};

export default renderWithBrowserRouter;
