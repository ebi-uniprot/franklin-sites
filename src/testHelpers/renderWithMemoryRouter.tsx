import { ReactElement, ReactNode } from 'react';
import { MemoryRouter, useLocation } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

// Inspired by https://testing-library.com/docs/example-react-router/

export const LocationDisplay = () => {
  const location = useLocation();
  return <div data-testid="location-display">{location.pathname}</div>;
};

const getWithLocation =
  (route: string) =>
  ({ children }: { children: ReactNode }) => (
    <MemoryRouter initialEntries={[route]}>
      {children}
      <LocationDisplay />
    </MemoryRouter>
  );

const renderWithMemoryRouter = (ui: ReactElement, { route = '/' } = {}) =>
  render(ui, { wrapper: getWithLocation(route) });

export default renderWithMemoryRouter;
