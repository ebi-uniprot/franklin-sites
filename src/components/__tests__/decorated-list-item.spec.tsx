import { screen, fireEvent, render } from '@testing-library/react';

import renderWithBrowserRouter from '../../testHelpers/renderWithBrowserRouter';
import renderWithMemoryRouter from '../../testHelpers/renderWithMemoryRouter';

import DecoratedListItem from '../decorated-list-item';

describe('DecoratedListItem component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <DecoratedListItem title="Title">Content</DecoratedListItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without title', () => {
    const { asFragment } = render(
      <DecoratedListItem>Content</DecoratedListItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render compact', () => {
    const { asFragment } = render(
      <DecoratedListItem title="Title" compact>
        Content
      </DecoratedListItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render compact without title', () => {
    const { asFragment } = render(
      <DecoratedListItem compact>Content</DecoratedListItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render inline', () => {
    const { asFragment } = render(
      <DecoratedListItem inline>Content</DecoratedListItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render link', () => {
    const { asFragment } = renderWithBrowserRouter(
      <DecoratedListItem to="/target">Content</DecoratedListItem>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render link and navigate when clicked', () => {
    renderWithMemoryRouter(
      <DecoratedListItem to="/target">Content</DecoratedListItem>
    );
    fireEvent.click(screen.getByTestId('background-link'));
    // verify location display is rendered
    expect(screen.getByTestId('location-display')).toHaveTextContent('/target');
  });
});
