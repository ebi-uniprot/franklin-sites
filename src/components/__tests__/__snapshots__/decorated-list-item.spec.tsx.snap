// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecoratedListItem component should render 1`] = `
<DocumentFragment>
  <div
    class="decorated-list-item"
  >
    <div
      class="decorated-list-item__title"
    >
      <span
        class="tiny"
      >
        Title
      </span>
    </div>
    <div
      class="decorated-list-item__content"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`DecoratedListItem component should render compact 1`] = `
<DocumentFragment>
  <div
    class="decorated-list-item decorated-list-item--compact"
  >
    <div
      class="decorated-list-item__title"
    >
      <span
        class="tiny"
      >
        Title
      </span>
    </div>
    <div
      class="decorated-list-item__content"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`DecoratedListItem component should render compact without title 1`] = `
<DocumentFragment>
  <div
    class="decorated-list-item decorated-list-item--compact"
  >
    <div
      class="decorated-list-item__content"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`DecoratedListItem component should render inline 1`] = `
<DocumentFragment>
  <div
    class="decorated-list-item decorated-list-item--inline"
  >
    <div
      class="decorated-list-item__content"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`DecoratedListItem component should render link and navigate when clicked 1`] = `
<DocumentFragment>
  <div
    class="decorated-list-item decorated-list-item--has-link"
  >
    <a
      aria-hidden="true"
      class="decorated-list-item__link"
      data-testid="background-link"
      href="/target"
      tabindex="-1"
    />
    <div
      class="decorated-list-item__content"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`DecoratedListItem component should render without title 1`] = `
<DocumentFragment>
  <div
    class="decorated-list-item"
  >
    <div
      class="decorated-list-item__content"
    >
      Content
    </div>
  </div>
</DocumentFragment>
`;
